src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4720,6 +4720,21 @@ public override IOperation VisitTypeParameterObjectCreation(ITypeParameterObject
             return HandleObjectOrCollectionInitializer(operation.Initializer, initializedInstance);
         }
 
+        public override IOperation VisitDynamicObjectCreation(IDynamicObjectCreationOperation operation, int? captureIdForResult)
+        {
+            PushArray(operation.Arguments);
+            ImmutableArray<IOperation> visitedArguments = PopArray(operation.Arguments);
```

Comment:
This is a giant code smell and at least deserves a comment. But probably Push/Pop should be deleted and replaced with something else.

Reply:
Mentioned offline that the names seem wrong. This should be called EvaluateItems or VisitAndPush, but also that if there were a helper method to combine these two things then the helper could assert that the stack is balanced, which I've found in a number of places in the code.

Reply:
@agocke I'll address this in a follow up PR, as we already have a PROTOTYPE comment on PushArray to revisit the method and determine if it's necessary. I'll probably end up renaming and adding a helper to assert stack size.

---
In reply to: [193550433](https://github.com/dotnet/roslyn/pull/27519#discussion_r193550433) [](ancestors = 193550433)